File: tests/leap_compat/kotlin/NewerConsumer.kt
package consumer

import ai.liquid.leap.HiddenStates
import ai.liquid.leap.LoraAdapterConfig
import ai.liquid.leap.ModelLoadingOptions
import ai.liquid.leap.ModelRunner

suspend fun adaptersAndEmbeddings(runner: ModelRunner): Array<FloatArray> {
    val adapters = listOf(LoraAdapterConfig(path = "adapter.gguf", scale = 0.5f))
    val options = ModelLoadingOptions(loraAdapters = adapters)
    runner.setLoraAdapters(options.loraAdapters.orEmpty())
    val states: HiddenStates = runner.hiddenStates(text = "hello", adapters = emptyList())
    val flat: FloatArray = states.data
    val tokens: Int = states.tokenCount
    val dimensions: Int = states.embeddingDim
    check(flat.size.toLong() == tokens.toLong() * dimensions)
    if (tokens > 0) check(states[0].size == dimensions)
    return states.toMatrix()
}

File: tests/leap_compat/swift/NewerConsumer.swift
import LeapSDK

func adaptersAndEmbeddings(_ runner: any ModelRunner) async throws -> [[Float]] {
  let adapters = [LoraAdapterConfig(path: "adapter.gguf", scale: 0.5)]
  let _: LiquidInferenceEngineOptions = LiquidInferenceEngineOptions(bundlePath: "model.bundle")
    .with(loraAdapters: adapters)
  try await runner.setLoraAdapters(adapters: adapters)
  let states: HiddenStates = try await runner.hiddenStates(text: "hello", adapters: [])
  let _: Int32 = states.tokenCount
  let _: Int32 = states.embeddingDim
  let _: [Float] = states.data.toFloatArray()
  let _: KotlinArray<KotlinFloatArray> = states.toMatrix()
  return (0..<states.tokenCount).map { states.get(token: $0).toFloatArray() }
}
